### ISSUE TITLE
Sorting of monthly dates in wrong order with write_ts if "transpose = T"

### ISSUE THREAD

**Issue by cgschwendt** [mbannert/tstools#75]
If monthly data is to be exported transposed using write_ts, october, november and december ("10", "11", "12") wrongly follow january("1") and precede february ("2")

Example: 
write_ts(dhu_all_series, fname = "dhu_zh",
         "csv",
         round_digits = 2,
         wide = T,
         transpose = T)

Screenshot:
![image](https://user-images.githubusercontent.com/35330634/36150227-423b6ed8-10c3-11e8-9be5-23d02727ed72.png)

I don't know to what extent this is problematic.

**Comment by HomoCodens** [comment]
Minimal example:
```
library(tstools)

ts1 <- ts(runif(30,0,abs(runif(1, 0, 50))),start=c(2010,1),frequency = 12)
ts2 <- ts(runif(30,0,abs(runif(1, 0, 80))),start=c(2010,1),frequency = 12)

tslist <- list(abc = ts1, xyz = ts2)

write_ts(tslist, fname = "non_transposed", wide = TRUE, timestamp_file = FALSE)
write_ts(tslist, fname = "transposed", transpose = TRUE, wide = TRUE, timestamp_file = FALSE)
```

The problem is that we transpose after the dates have been converted to strings so they are sorted wrong.
This also breaks read_ts because it will wrongly try to convert regular ts to xts and stumble over another bug (#76).

**Comment by mbannert** [comment]
This is important, even though the I/O stuff is not used a lot yet. Rather fix it soon. 
I think we should create a new milestone: bugfix release CRAN or so and add some task to it.

**Comment by HomoCodens** [comment]
OK actually the problem is in formatNumericDate. Specifically, when no date_format is provided, the month is not zero-padded and sorted wrong.

```
> a <- sprintf("%d-%d", 2018, 01)
> a
[1] "2018-1"
> b <- format(as.Date(sprintf("%d-%d-01", 2018, 1)), "%Y-%m")
> b
[1] "2018-01"
``` 
Should we still inclide GH-78?

**Comment by HomoCodens** [comment]
>This is important, even though the I/O stuff is not used a lot yet. Rather fix it soon.
>I think we should create a new milestone: bugfix release CRAN or so and add some task to it.
Should we wait for more overlooked bugs to accumulate or push fixes as quickly as possible?

**Comment by mbannert** [comment]
Good question. I think once we passed the initial check, the next fix is easy to upload to CRAN. 
I did the same with kofdata which was not a problem. I would rather fix this soon. I mean we already got two bugs. I think you fixed #75 as well, so that's two already. Let's have a quick briefing tomorrow, add a couple of tests and get it done. I suggest, we fix

- this one
- #75
- unit test with monthly zeros

and then go again.
